spec/cop/sorbet/forbid_untyped_struct_props_spec.rb

```diff
@@ -0,0 +1,93 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+require_relative '../../../lib/rubocop/cop/sorbet/forbid_superclass_const_literal'
+
+RSpec.describe(RuboCop::Cop::Sorbet::ForbidUntypedStructProps, :config) do
+  subject(:cop) { described_class.new(config) }
+
+  it 'adds offense when const is T.untyped' do
+    expect_offense(<<~RUBY)
+      class MyClass < T::Struct
+        const :foo, T.untyped
+                    ^^^^^^^^^ Struct props cannot be T.untyped
+      end
+    RUBY
+  end
+
+  it 'adds offense when const is T.nilable(T.untyped)' do
+    expect_offense(<<~RUBY)
+      class MyClass < T::Struct
+        const :foo, T.nilable(T.untyped)
+                    ^^^^^^^^^^^^^^^^^^^^ Struct props cannot be T.untyped
```

Comment:
I would also add a recursive test:

```ruby
class A < T::Struct
  const :foo, T.nilable(T.nilable(T.untyped))
end
```

Reply:
Added a test for nested `T.nilable` and handled that case in the cop.